src/ui/RealmBrowser/Content/CreateObjectDialog/index.tsx

```diff
@@ -87,16 +87,18 @@ class CreateObjectDialogContainer extends React.PureComponent<
   ) {
     // TODO: Initialize the values based on their property
     if (
-      propertyName === 'uuid' &&
+      propertyName === '_id' &&
       propertyName === primaryKey &&
-      property.type === 'string'
+      property.type === 'object id'
     ) {
-      return uuid();
+      return new ObjectId();
     } else if (property.type === 'list') {
       // If a list is optional, it refers to the type of the elements
       return [];
     } else if (property.optional) {
       return null;
+    } else if (property.type === 'object id') {
+      return new ObjectId();
```

Comment:
I agree with this change, but doesn't that make the first condition meaningless? I.e. if we'll generate object id for all properties of `object id` type, then what does it matter if they're called `_id`?

Reply:
Any optional/nullable property defaults to `null` in the if clause before the one mentioned:
https://github.com/realm/realm-studio/blob/cbcf32d85f5989143b82c3bc1310df47fb466701/src/ui/RealmBrowser/Content/CreateObjectDialog/index.tsx#L98-L99
The first condition opts out of defaulting to a `null` value if the primary key is optional.

I had originally removed the first if clause, until I realised a primary key can be optional/nullable. I'm not sure in what scenario an optional primary key would make sense, and would argue that this option could only end up confusing developers - but it's currently possible.

Reply:
I guess that actually didn't answer the actual question:
> then what does it matter if they're called _id?

In the first if clause I should probably only check if it's a primaryKey of type `'object id'`.

Reply:
I see, so if an object id field is nullable, we only want to generate a value if that field is a PK? I guess that's fine although I think it might be a good idea to generate it either way - it's always easier to assign a generated value to null than to assign null value to some random object id.

Reply:
I've added a “generate button” to the `ObjectIdControl`, for easy random generation of ObjectIds, partly to keep the default nulling of optional values.